sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/QueueServiceAPITests.groovy

```diff
@@ -0,0 +1,182 @@
+        queueClient.enqueueMessage("Expecting an exception")
+        then:
+        def e = thrown(StorageErrorException)
+        QueueTestHelper.assertExceptionStatusCodeAndMessage(e, 404, StorageErrorCode.QUEUE_NOT_FOUND)
+    }
+
+    def "Create queue from queue service client"() {
+        when:
+        def queueClientResponse = primaryQueueServiceClient.createQueue(testResourceName.randomName("queue", 16))
+        def enqueueMessageResponse = queueClientResponse.value().enqueueMessage("Testing service client creating a queue")
+        then:
+        QueueTestHelper.assertResponseStatusCode(queueClientResponse, 201)
+        QueueTestHelper.assertResponseStatusCode(enqueueMessageResponse, 201)
+    }
+
+    def "Create queues with duplicate name from queue service client"() {
+        given:
+        def queueName = testResourceName.randomName("queue", 16)
+        when:
+        primaryQueueServiceClient.createQueue(queueName)
+        def queueClientResponse = primaryQueueServiceClient.createQueue(queueName)
+        then:
+        QueueTestHelper.assertResponseStatusCode(queueClientResponse, 204)
+    }
+
+    @Unroll
+    def "Create queue with invalid name from queue service client"() {
+        when:
+        primaryQueueServiceClient.createQueue(queueName)
+        then:
+        def e = thrown(StorageErrorException)
+        QueueTestHelper.assertExceptionStatusCodeAndMessage(e, statusCode, errMesage)
+        where:
+        queueName      | statusCode | errMesage
+        "a_b"          | 400        | StorageErrorCode.INVALID_RESOURCE_NAME
+        "-ab"          | 400        | StorageErrorCode.INVALID_RESOURCE_NAME
+        "a--b"         | 400        | StorageErrorCode.INVALID_RESOURCE_NAME
+        // null | 400 | "InvalidResourceName" TODO: Need to fix the RestProxy before having null parameter
+        "Abc"          | 400        | StorageErrorCode.INVALID_RESOURCE_NAME
+        "ab"           | 400        | StorageErrorCode.OUT_OF_RANGE_INPUT
+        "verylong" * 8 | 400        | StorageErrorCode.OUT_OF_RANGE_INPUT
+    }
+
+    @Unroll
+    def "Create queue maxOverload from queue service client"() {
+        when:
+        def queueClientResponse = primaryQueueServiceClient.createQueue(testResourceName.randomName("queue", 16), metadata)
+        def enqueueMessageResponse = queueClientResponse.value().enqueueMessage("Testing service client creating a queue")
+        then:
+        QueueTestHelper.assertResponseStatusCode(queueClientResponse, 201)
+        QueueTestHelper.assertResponseStatusCode(enqueueMessageResponse, 201)
+        where:
+        metadata                                       | _
+        null                                           | _
+        Collections.singletonMap("metadata", "value")  | _
+        Collections.singletonMap("metadata", "va@lue") | _
+    }
+
+    def "Create queue with invalid metadata from queue service client"() {
+        given:
```

Comment:
Any comments I left on create queue in the QueueClient tests should apply here. ALthough I might even assert that we don't need to duplicate all those tests since we are just passing along the api call to the QueueClient anyway.

Reply:
Same for delete queue tests

Reply:
We currently has code coverage. It is better to have all API covered.